back-end/src/main/java/box/BoxGetter.java

```diff
@@ -48,6 +48,37 @@
         return data;
     }
 
+    public ArrayList<ArrayList<ArrayList<String>>> fetchCubeData(Canvas c, Box newBox, Box oldBox, ArrayList<String> predicates)
+            throws Exception {
```

Comment:
when adding a new indexer, you are not expected to modify BoxGetters. But as we discussed, this is my bad -- the current `getDataFromRegion` call lacks two important arguments `newBox` and `oldBox`. Adam has a [fix in his branch](https://github.com/tracyhenry/Kyrix/blob/d5937221cb0452d4ca12d09ee79db31ef59e0ffe/back-end/src/main/java/index/Indexer.java#L29) , but you can take a look at it and copy it to your branch first. We can worry about merge later. 

Then, what's expected is you don't need to modify/add `BoxGetter` or `BoxRequestHandler`.